pkg/storage/snapshot/source.go

```diff
@@ -407,6 +408,22 @@ func (s *vmSnapshotSource) Spec() (snapshotv1.SourceSpec, error) {
 	}, nil
 }
 
+func (s *vmSnapshotSource) handleVolumes(vmCpy *snapshotv1.VirtualMachine) error {
+	vmi, exists, err := s.controller.getVMI(s.vm)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("can't get vm revision, vmi doesn't exist")
+	}
+	// Include only if live-update is persistent
+	if reflect.DeepEqual(vmi.Spec.Volumes, s.vm.Spec.Template.Spec.Volumes) && !reflect.DeepEqual(s.vm.Spec.Template.Spec.Volumes, vmCpy.Spec.Template.Spec.Volumes) {
```

Comment:
check out `apiequality.Semantic.DeepEqual()`

But not sure this is exactly what we want. For example, say the user adds one hotplug disk and one regular disk to the VM.

I think the right thing to do would be to iterate the volumes in the VMI and check if they are in the VM, if so add them to the snapshot